ADAL/src/ADAuthenticationParameters+Internal.m

```diff
@@ -168,6 +168,150 @@ + (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
         *error = adError;
     }
     return nil;
+
+}
+
++ (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
+                                       error:(ADAuthenticationError * __autoreleasing *)error;
+{
+    __block ADAuthenticationError* adError = nil;
+    NSDictionary *parameters = nil;
+    
+    if ([NSString adIsStringNilOrBlank:headerContents])
+    {
+        adError = [self invalidHeader:headerContents];
+    }
+    else
+    {
+        NSString *bearerChallendge = [self extractBearerChallenge:headerContents];
+        if (!bearerChallendge) {
+            adError = [self invalidHeader:headerContents];
+        } else {
+            parameters = [self parseBearerChallenge:bearerChallendge error:&adError];
+        }
+    }
+    
+    if (error)
+    {
+        *error = adError;
+    }
+    
+    return parameters;
+}
+
+#pragma mark - Private
+
++ (NSUInteger)quotedStringLastIndex:(NSString *)headerContents startIndex:(NSUInteger)startIndex
+{
+    NSUInteger lastIndex = startIndex;
+    
+    unichar quotedChar = [headerContents characterAtIndex:startIndex];
+    if (quotedChar == '"' || quotedChar == '\'')
+    {
+        NSUInteger nextIndex = startIndex + 1;
+        while (nextIndex < headerContents.length) {
+            unichar c = [headerContents characterAtIndex:nextIndex];
+            
+            if (c == quotedChar) {
+                lastIndex = nextIndex;
+                break;
+            }
+            
+            nextIndex++;
+        }
+    }
+    
+    return lastIndex;
+}
+
++ (NSString *)extractBearerChallenge:(NSString *)headerContents
+{
+    NSRange range = [headerContents rangeOfString:@"Bearer "];
+    
+    if (range.location == NSNotFound)
+    {
+        return nil;
+    }
+    
+    NSUInteger nextIndex = range.location + range.length;
+    NSUInteger lastChallengeIndex = NSNotFound;
+    
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)